### PR TITLE
fix(external table):get table and external table together without exception

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-bug

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java [hunk 1]
```diff
@@ -126,11 +142,29 @@ public List<Table> list(@NonNull @Valid QueryTableParams params) throws SQLExcep
...
+                            DBObjectType.EXTERNAL_TABLE, tableExtension);
+                }
             }
-            return tables;
         }
+        return tables;
+    }
+
+    @SkipAuthorize("permission check inside")
+    public boolean checkExternalTableSupported(@NonNull DialectType dialectType, @NonNull Connection conn)
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java:155]: checkExternalTableSupported -> isExternalTableSupported

check prefix usually lead to an exception and 'is' prefix usually lead to a boolean

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java [hunk 2]
```diff
@@ -224,4 +258,17 @@ private List<Table> entitiesToModels(Collection<DBObjectEntity> entities, Databa
         return tables;
     }
 
+    private String initDBSupportExternalTableVersion(@NonNull DialectType dialectType) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java:261]: we may add a new method on `VersionDiffConfigService`, such as `VersionDiffConfigService#getSupportFeatures`, btw, the method with this name has been existed, we may create another one with different parameters, just like this:
```java
public List<OBSupport> getSupportFeatures(String version, ConnectType connectType) {
 ...
}
```
please notice the duplicated code, we may reuse some codes with the current method named `getSupportFeatures`

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java [hunk 3]
```diff
@@ -126,11 +142,29 @@ public List<Table> list(@NonNull @Valid QueryTableParams params) throws SQLExcep
...
             }
-            return tables;
         }
+        return tables;
+    }
+
+    @SkipAuthorize("permission check inside")
+    public boolean checkExternalTableSupported(@NonNull DialectType dialectType, @NonNull Connection conn)
+            throws SQLException {
+        String databaseProductVersion = conn.getMetaData().getDatabaseProductVersion();
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java:157]:
```java
 InformationExtensionPoint point =
                ConnectionPluginUtil.getInformationExtension(connectionSession.getDialectType());
        point.getDBVersion()
```
these codes may help you get the version

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java [hunk 4]
```diff
@@ -103,6 +113,12 @@ public class TableService {
     @Autowired
     private DBResourcePermissionHelper dbResourcePermissionHelper;
 
+    @PostConstruct
+    public void init() {
+        OB_MYSQL_SUPPORT_EXTERNAL_TABLE_VERSION = initDBSupportExternalTableVersion(DialectType.OB_MYSQL);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java:118]: check it every time when we calling `list` method, don't init it in postconstruct

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java [hunk 5]
```diff
@@ -58,6 +67,19 @@ protected void preDelete(@NonNull Set<Long> toBeDeletedIds) {
         userPermissionRepository.deleteByPermissionIds(permissionIds);
     }
 
+    @Override
+    public boolean supports(@NonNull DialectType dialectType, @NonNull Connection connection) {
+        boolean externalTableSupported;
+        try {
+            externalTableSupported = tableService.checkExternalTableSupported(dialectType, connection);
+        } catch (SQLException e) {
+            log.warn("checkSupportObtainExternalTableList failed, dialectType:{}, connection:{}", dialectType,
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java:76]: don't print the connection, there are no more info can be printing,
```java
log.warn("Failed to check if external table supported", e);
```

this log printing will be okay

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java [hunk 6]
```diff
@@ -41,13 +46,17 @@
...
 public class DBExternalTableSyncer extends AbstractDBObjectSyncer<TableExtensionPoint> {
 
     @Autowired
     private PermissionRepository permissionRepository;
 
     @Autowired
     private UserPermissionRepository userPermissionRepository;
+    @Autowired
+    @Lazy
+    private TableService tableService;
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java:59]: why `Lazy`?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java (hunk 4 above)
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java:118]: ok

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java (hunk 5 above)
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java:76]: ok

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java (hunk 6 above)
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java:59]: loop dependency

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java (hunk 3 above)
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java:157]: ok

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java (hunk 1 above)
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java:155]: ok

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java (hunk 6 above)
Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java:59]: we may solve this problem as follow:
1. add a new method  in VersionDiffService, named `isExternalSupported`
2. ref this method in bot tableservice and here

circle ref loop is not allowed

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java [hunk 7]
```diff
@@ -58,6 +65,12 @@ protected void preDelete(@NonNull Set<Long> toBeDeletedIds) {
         userPermissionRepository.deleteByPermissionIds(permissionIds);
     }
 
+    @Override
+    public boolean supports(@NonNull DialectType dialectType, @NonNull Connection connection) {
+        return versionDiffConfigService.isExternalTableSupported(dialectType, connection)
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/object/DBExternalTableSyncer.java:70]: we may use try-catch block

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by PeachThinking — DISMISSED** [review]
lgtm

**Review by MarkPotato777 — DISMISSED** [review]

File: server/odc-service/src/main/java/com/oceanbase/odc/service/feature/VersionDiffConfigService.java [hunk 8]
```diff
@@ -135,6 +140,23 @@ public List<OBSupport> getSupportFeatures(ConnectionSession connectionSession) {
         return obSupportList;
     }
 
+    public boolean isExternalTableSupported(@NonNull DialectType dialectType, @NonNull Connection conn) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/feature/VersionDiffConfigService.java:143]: involve unnecessary metadb query for each 'listTables' API, may:

1. VersionDiffConfigService should not depends on Connection and Plugin, it is a configuraion module which is designed for depended by other modules
2. externalTableSupported may initialized while ConnectionSession created
3. VersionDiffConfigService  may involve cache for better performance

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]
lgtm

**Pull request merged**